oat-sa/tao-core
# Feature/tao 10457/implement class filter

**related to:** https://oat-sa.atlassian.net/browse/TAO-10457
**Depends of:** https://github.com/oat-sa/tao-core-ui-fe/pull/178

**description:** `searchModal` component on `tao-core-ui-fe` now supports a class filter input, so `search` component on `tao-core` needs to provide the `rootClassUri` of current `context` . Also a new `criterias` property has been created on `searchStore` to save class filter input value and any future criteria input value, so `tao-core` has been updated to correctly manage it.

**How to test**
To be able to test this PR locally you need to setup your TAO instance to manage the new Advanced search feature. To do so you need to:

On your root `composer.json` set the following dependencies:
```
    "oat-sa/tao-core": "dev-feature/TAO-10457/implement-class-filter",
    "oat-sa/generis": "dev-feature/TAO-10203-advanced-search as 12.33.2",
    "oat-sa/extension-tao-dac-simple": "dev-feature/TAO-10203-advanced-search",
    "oat-sa/lib-tao-elasticsearch": "dev-feature/TAO-10203-advanced-search as 1.1.1"
```
run `composer install`
if you work with taoDockerize run `taoDockerize install` (database will be reset)
If you do not user taoDockerize run `php tao/scripts/taoUpdate.php`
Notice that because of its dependency, on `views/package.json` `@oat-sa/tao-core-ui` needs to be pointing to `feature/TAO-10457/implement-class-filter` your usual way to do it (linking package to local repository, pointing package.json to origin branch...).

Now `searchModal` component should be used on search trigger instead of previous search component. Notice that advanced search syntax is not enabled if you do not set up an elastic search server, which is not required to test this feature.

Please let me know if you need more details about how to test the feature.

![class-filter](https://user-images.githubusercontent.com/14041944/92386677-8159b000-f114-11ea-86ea-f739cdb59bea.gif)
